### PR TITLE
fix: fix negative balances

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

we were deducting erc20 values from eth, causing charts to go negative 

i added a bunch of utility functions that turned up not being required and associated tests

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [ ] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [ ] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.

## Testing

Please outline all testing steps

1. Pull branch locally and run `yarn` to install new deps
2. etc...

## Screenshots (if applicable)

Screen Shot 2021-11-18 at 3 17 59 PM

### CODE REVIEW

File: src/context/WalletProvider/NativeWallet/components/NativeLoad.tsx [hunk 1]
```diff
@@ -98,9 +98,10 @@ export const NativeLoad = () => {
       </ModalHeader>
       <ModalBody>
         <VStack mx={-4} spacing={0}>
-          {wallets.map(wallet => {
+          {wallets.map((wallet, i) => {
```
Comment (0xdef1cafe) [inline src/context/WalletProvider/NativeWallet/components/NativeLoad.tsx:101]: fixes react console warning on native wallet load

File: src/hooks/useBalanceChartData/useBalanceChartData.ts [hunk 2]
```diff
@@ -46,7 +47,7 @@ export const priceAtBlockTime: PriceAtBlockTime = ({ time, assetPriceHistoryData
 }
 
 type CryptoBalance = {
-  [k: CAIP19]: number // map of asset to base units
+  [k: CAIP19]: BigNumber // map of asset to base units
```
Comment (0xdef1cafe) [inline src/hooks/useBalanceChartData/useBalanceChartData.ts:50]: changing this to big number caught a bug with large erc20 sends and is more accurate

File: src/hooks/useBalanceChartData/useBalanceChartData.ts [hunk 3]
```diff
@@ -137,7 +138,29 @@ const caip19FromTx = (tx: Tx): CAIP19 => {
 
   const extra = contractType ? { contractType, tokenId } : undefined
   const assetCAIP19 = caip19.toCAIP19({ chain, network, ...extra })
-  return assetCAIP19
+  return assetCAIP19.toLowerCase()
+}
+
+export const buyAssetCAIP19FromTx = (tx: Tx, portfolioAssets: PortfolioAssets): string => {
```
Comment (0xdef1cafe) [inline src/hooks/useBalanceChartData/useBalanceChartData.ts:144]: unused

File: src/hooks/useBalanceChartData/useBalanceChartData.ts [hunk 4]
```diff
@@ -137,7 +138,29 @@ const caip19FromTx = (tx: Tx): CAIP19 => {
...
+  if (!tradeDetails) return ''
+  const buyAssetSymbol = tradeDetails.buyAsset
+  const buyAsset: Asset | undefined = Object.values(portfolioAssets).find(
+    ({ symbol }) => symbol === buyAssetSymbol
+  )
+  if (!buyAsset) return ''
+  return buyAsset.caip19
+}
+
+export const sellAssetCAIP19FromTx = (tx: Tx, portfolioAssets: PortfolioAssets): string => {
```
Comment (0xdef1cafe) [inline src/hooks/useBalanceChartData/useBalanceChartData.ts:155]: unused

File: src/hooks/useBalanceChartData/useBalanceChartData.ts [hunk 5]
```diff
@@ -189,14 +214,21 @@ const fiatBalanceAtBucket: FiatBalanceAtBucket = ({
     const assetPriceHistoryData = priceHistoryData[caip19]
     const price = priceAtBlockTime({ assetPriceHistoryData, time })
     const portfolioAsset = portfolioAssets[caip19]
-    if (!portfolioAsset) return acc
+    if (!portfolioAsset) {
```
Comment (0xdef1cafe) [inline src/hooks/useBalanceChartData/useBalanceChartData.ts:217]: a warning to catch the rare case this happens with useful debugging info

File: src/pages/Dashboard/components/AccountList/AccountList.tsx [hunk 6]
```diff
@@ -16,7 +17,6 @@ export const AccountList = ({ loading }: { loading?: boolean }) => {
   const assets = useSelector((state: ReduxState) => state.assets)
   const marketData = useSelector((state: ReduxState) => state.marketData.marketData)
   const { balances, totalBalance } = usePortfolio()
-  const emptyAccounts = new Array(5).fill(null)
```
Comment (0xdef1cafe) [inline src/pages/Dashboard/components/AccountList/AccountList.tsx:19]: this was causing a rerender each time

**Review by kaladinlight — APPROVED** [review]

**Review by natven — APPROVED** [review]
🌮

**Review by cjthompson — APPROVED** [review]

**Pull request merged**